include/seqan/bam_io/bam_index_bai.h

```diff
@@ -298,22 +299,47 @@ jumpToRegion(FormattedFile<Bam, Input, TSpec> & bamFile,
         readRecord(record, bamFile);
 
         // std::cerr << "record.beginPos == " << record.beginPos << "\n";
-         int32_t endPos = record.beginPos + getAlignmentLengthInRef(record);
         if (record.rID != refId)
             continue;  // Wrong contig.
-        if (!hasAlignments && record.beginPos <= posEnd && pos <= endPos)
+
+        if (record.beginPos <= regionStart)
         {
-            // Found a valid alignment.
-            hasAlignments = true;
             offset = *candIt;
         }
 
-        if (record.beginPos >= posEnd)
+        if (record.beginPos >= regionEnd)
             break;  // Cannot find overlapping any more.
     }
 
-    if (offset != std::numeric_limits<uint64_t>::max())
-        setPosition(bamFile, offset);
+    if (offset == std::numeric_limits<uint64_t>::max())
+        return true; // Finding no overlapping alignment is not an error, hasAlignments is false.
+
+    // Now only the right most offset was found but it is not ensured that the
+    // alignment at the start of the offset lies inside the region.
+    // Therefore we must advance further or until the region ends.
+    setPosition(bamFile, offset);
+
+    while (!atEnd(bamFile))
+    {
+        auto seek_pos = position(bamFile); // store current pos to reset bamFile if necessary
+        readRecord(record, bamFile);
+
+        int32_t recordEndPos = record.beginPos + getAlignmentLengthInRef(record);
+
+        if (record.beginPos < regionEnd)
+        {
+            if (recordEndPos > regionStart)
+            {
+                hasAlignments = true;
+                setPosition(bamFile, seek_pos);
+                break;
+            }
+        }
+        else
+        {
+            break;
+        }
+    }
```

Comment:
I would remove the comment below it's redundant and has no meaning here.

Reply:
True.